### PR TITLE
Disable Stylecop on debug config

### PR DESCRIPTION
**Pull request by teinarss**
Disables Stylecop in the default build configuration to make it not as cumbersome when developing and testing new code,
added a new configuration for stylecop checks.

### CODE REVIEW

File: Directory.Build.props
Comment (pchote) [inline Directory.Build.props]: `'$(Configuration)'!='StyleCop'`?

Also, does `StyleCop` behave like `Debug` wrt everything that isn't stylecop, or like `Release`? IMO it would be good if the name could something be like `Debug-Strict` to make it clearer how it behaves.

**Comment by teinarss** [comment]
Fixed.

File: Directory.Build.props
Comment (abcdefg30) [inline Directory.Build.props]: This Condition is inside an element with the same condition already, so do we need this?

**Comment by teinarss** [comment]
Updated.

**Review by abcdefg30 — DISMISSED** [review]

**Comment by RoosterDragon** [comment]
Not sure how feasible this is so not a blocker and just a thought - but would it be possible to get Stylecop enabled as warnings for local dev but errors in the build pipeline? I would find that more ergonomic - I can run things while I'm trying stuff out but still have a warning to remind me to fix it when I'm ready to clean things up.

**Comment by teinarss** [comment]
Updated after discussion on discord

**Comment by pchote** [comment]
With the new approach, would it be simpler to drop the Debug-Strict configuration and instead add `-warnaserror` to the `dotnet build` line in `make check`?

**Comment by teinarss** [comment]
Updated.

File: make.ps1
Comment (pchote) [inline make.ps1]: nit: double spaces!

**Comment by teinarss** [comment]
Fixed

**Review by pchote — APPROVED** [review]
LGTM :+1:

This has completely changed since the version @abcdefg30 reviewed, so a second review would be good before merging.

**Comment by abcdefg30** [comment]
My 👍 stands (again).

**Pull request merged**
